src/_pytest/python_api.py

```diff
@@ -154,6 +154,12 @@ def __eq__(self, actual) -> bool:
 
         return ApproxBase.__eq__(self, actual)
 
+    def _approx_scalar(self, x):
```

Comment:
Also, any reason not to do it in the base `_approx_scalar`?

Reply:
I *think* there was a reason but I can't seem to remember it :o I'll try it out :+1:

Reply:
I moved the logic to the base `_approx_scalar` and it seems to work fine. Thanks for the push :+1: